sklearn/datasets/_samples_generator.py

```diff
@@ -100,7 +100,9 @@ def make_classification(n_samples=100, n_features=20, n_informative=2,
         ``weights`` exceeds 1.
 
     flip_y : float, optional (default=0.01)
-        The fraction of samples whose class is assigned randomly. Larger
+        The fraction of samples whose class is assigned randomly. If you
+        prefer the output dataset to have all the class labels you specified
+        at n_classes, you should set flip_y to 0. Larger
```

Comment:
I don't think having flip_y > 0 necessarily implies that there will be less than `n_classes` in `y` though, right?

How about e.g.:

> Note that setting `flip_y > 0` might lead to less than `n_classes` in `y` in some cases.

Reply:
Right, it's not a guarrentee, but it could happen. As discussed in #16789 ?

Reply:
yes it could happen. But the current formulation suggests that it will always happen when `flip_y > 0`. At least that's how I read it. I feel like the proposal in my comment above is less ambiguous

Reply:
ok, yeah, i see your point. I will use your comment there. Thanks